lib/importers/lastPass.js

```
@@ -0,0 +1,181 @@
+  {
+    let currentChar = fileContents[i];
+    switch (currentChar)
+    {
+      case "\"":
+        if (!currentlyQuoted)
+          currentlyQuoted = true;
+        else if (i + 1 < fileContents.length &&
+                 fileContents[i + 1] == "\"")
+        {
+          currentValue += currentChar;
+          i++;
+        }
+        else
+          currentlyQuoted = false;
+        break;
+      case ",":
+        if (!currentlyQuoted)
+        {
+          currentEntry.push(currentValue);
+          currentValue = "";
+        }
+        else
+          currentValue += currentChar;
+        break;
+      case "\n":
+        if (!currentlyQuoted)
+        {
+          currentEntry.push(currentValue);
+          currentValue = "";
+          if (currentEntry.length != expectedValueCount)
+          {
+            throw new Error(
+              "Line " + currentLine + ": Wrong number of values for entry. " +
+              "Saw " + currentEntry.length + ", but expected " +
+              expectedValueCount + "."
+            );
+          }
+          entries.push(currentEntry);
+          currentEntry = [];
+        }
+        else
+          currentValue += currentChar;
+        currentLine++;
+        break;
+      default:
+        currentValue += currentChar;
+    }
+  }
+
+  if (currentlyQuoted)
+  {
+    throw new Error(
+      "Syntax error, quotation mark was opened but never closed!"
+    );
+  }
+
+  return entries;
+}
+exports.parseCSV = parseCSV;
+
+function import_(data, setRaw, setSite, setPassword)
+{
+  return Promise.resolve().then(() =>
+  {
+    // Quick sanity check, does this file have the right header?
+    if (!data.trim().startsWith(header + "\n"))
+      throw "unknown-data-format";
+
+    let mergeActions = [];
+
+    let seenSites = new Set();
+    let entries;
+    try
+    {
+      entries = parseCSV(decodeHTMLEntities(data));
+    }
+    catch (e)
+    {
+      throw "unknown-data-format";
```

Throwing "unknown-data-format" will suppress any error output and simply try the next importer. We already verified however that this is a LastPass CSV, so parsing errors are unexpected and should be surfaced. So instead of asking "why isn't my file being recognized?" users should be able to ask about a specific error (or maybe even research it themselves).